yt/frontends/enzo_e/fields.py

```diff
@@ -143,17 +143,34 @@ def setup_particle_mass_field(self, ptype):
                 constants = (constants,)
             names = [c[0] for c in constants]
 
+        group_list = nested_dict_get(
+            self.ds.parameters, ("Particle", ptype, "group_list"), default=()
+        )
+
         if "mass" in names:
-            val = constants[names.index("mass")][2]
-            val = self.ds.quan(val, self.ds.mass_unit)
-            if self.ds.cosmological_simulation:
+            # Prior to the addition of the is_gravitating group list,
+            # particle masses were stored as densities if the simulation
+            # was cosmological.
+            pfield = "mass"
+            is_density = (
+                "is_gravitating" not in group_list and self.ds.cosmological_simulation
+            )
+        elif "density" in names:
+            pfield = "density"
+            is_density = True
+        else:
+            pfield = None
+
+        if pfield is not None:
+            val = constants[names.index(pfield)][2] * self.ds.mass_unit
+            if is_density:
                 val = val / self.ds.domain_dimensions.prod()
```

Comment:
This only works if the domain width is 1 in all dimensions in code units (which is the case for cosmological simulations). In general, `val` should be multiplied by the root level cell volume.

Reply:
Will fix. Thanks!